cmd/crc/cmd/start.go

```diff
@@ -96,3 +108,44 @@ func validateStartFlags() error {
 	}
 	return nil
 }
+
+func getPullSecretFileContent() (string, error) {
+	var (
+		pullsecret string
+		err        error
+	)
+	// Check if pull secret is stored in cache then read it and return.
+	pullsecret, err, pullSecretFilePresent := ps.GetPullSecretFromCache(filepath.Join(constants.MachineCacheDir, constants.PullSecretFile))
+	if err != nil {
+		return "", err
+	}
+
+	if !pullSecretFilePresent {
+		// In case user doesn't provide a file in start command or in config then ask for it.
+		if crcConfig.GetString(config.PullSecretFile.Name) == "" {
+			pullsecret, err = input.PromptUserForSecret("Image pull secret", fmt.Sprintf("Copy it from %s", constants.DefaultPullSecretURL))
+			// This is just to provide a new line after user enter the pull secret.
+			fmt.Println()
+			if err != nil {
+				return "", err
+			}
+		} else {
+			// Read the file content
+			data, err := ioutil.ReadFile(crcConfig.GetString(config.PullSecretFile.Name))
+			if err != nil {
+				return "", err
+			}
+			pullsecret = string(data)
+		}
+	}
+	if err := validation.ImagePullSecret(pullsecret); err != nil {
+		return "", err
+	}
+
+	// Add pull secret to cache.
+	if err := ps.StorePullSecretToCache(pullsecret, constants.MachineCacheDir, constants.PullSecretFile, 0755); err != nil {
```

Comment:
@cfergeau there is a reason since `ps.StorePullSecretToCache` is actually check the if parent directory available or not and create if not available but later one just need to read the file with a given file path. `0755` is there if the directory not present and we are creating all the directory with this permission.

Reply:
Ah I think I'd favour passing 0644, and using `err := os.MkdirAll(destDir, mode | 0111)` (or just hardcode 0755/0644 in that function).
However, now that we talk about it, I don't think the pull secret should be readable by anyone with an account on the system, I'd use 0700 and 0600 to keep it secret from other users.